lib/chef/resource/hostname.rb

```diff
@@ -71,6 +89,12 @@ class Hostname < Chef::Resource
         description: "Determines whether or not Windows should be reboot after changing the hostname, as this is required for the change to take effect.",
         default: true
 
+      property :windows_domain_username, String,
```

Comment:
Let's just call this `domain_user` and `domain_password` and set the description "Windows only: whatever whatever whatever"

Reply:
Also domain_password should be sensitive: true so we don't leak that